File: ruby/src/Behavioral/Mediator/DvdLowercaseTitle.rb


#//DvdLowercaseTitle- One of Two Concrete Colleagues or Mediatees

require "Behavioral/Mediator/DvdTitle"

class DvdLowercaseTitle < DvdTitle

	attr_accessor :lowercasetitle, :title

	def initialize( title, dvdMediator)
		@LowercaseTitle = nil
		if title.is_a? DvdTitle
			setTitle( title.getTitle() )
		else 
			setTitle(title)
		end
		resetTitle()
		@dvdMediator = dvdMediator
		@dvdMediator.setLowercase(self)
	end

	def resetTitle (title = nil)
 		if title
			setTitle(title)
		end
		setLowercaseTitle(getTitle().downcase())
	end

	def setSuperTitleLowercase()
 		setTitle(getLowercaseTitle())
		@dvdMediator.changeTitle(self)
	end

	def getLowercaseTitle()
 		return @lowercasetitle
	end

	def setLowercaseTitle(aLowercaseTitle)
 		@lowercasetitle = aLowercaseTitle
	end

end

File: csharp/src/Behavioral/Mediator/DvdLowercaseTitle.cs
namespace Behavioral.Mediator {


//DvdLowercaseTitle.java - One of Two Concrete Colleagues or Mediatees

public class DvdLowercaseTitle : DvdTitle {
   private string LowercaseTitle;
   private DvdMediator dvdMediator;
    
   public DvdLowercaseTitle(string title, DvdMediator dvdMediator) {
       this.setTitle(title);
       resetTitle();
       this.dvdMediator = dvdMediator;
       this.dvdMediator.setLowercase(this);
   }    
   
   public DvdLowercaseTitle(DvdTitle dvdTitle, 
                            DvdMediator dvdMediator) 
       : this(dvdTitle.getTitle(), dvdMediator)
	{
   }     
   
   public void resetTitle() {
       this.setLowercaseTitle(this.getTitle().ToLower());
   }
   public void resetTitle(string title) {
       this.setTitle(title);
       this.resetTitle();
   }
   
   public void setSuperTitleLowercase() {
       this.setTitle(this.getLowercaseTitle());
       dvdMediator.changeTitle(this);
   }
   
   public string getLowercaseTitle() {
       return LowercaseTitle;
   }
   private void setLowercaseTitle(string LowercaseTitle) {
       this.LowercaseTitle = LowercaseTitle;
   }
}

}
